_source/_posts/2022-10-01-spring-vault.md

```diff
@@ -0,0 +1,639 @@
+
+  * Endpoint ignored these unrecognized parameters: [display_name entity_alias
+  explicit_max_ttl num_uses period policies renewable ttl type]
+
+Key                  Value
+---                  -----
+token                hvs.CAESIKd9pYyc9xesiqmwvepdGiHlPEB53By...
+token_accessor       2pzRXmGhxChobbMtqeM5zZzM
+token_duration       768h
+token_renewable      true
+token_policies       ["default" "vault-demo-app-policy"]
+identity_policies    []
+policies             ["default" "vault-demo-app-policy"]
+```
+
+**NOTE**: I could not find documentation about the warning _Endpoint ignored these unrecognized parameters_. It seems `vault CLI` is sending default parameters not required by the target API in this case. The command equivalent API call can be displayed using the `-output-curl-string` flag after the subcommand, for example:
+
+```bash
+vault token create -output-curl-string -policy=vault-demo-app-policy
+```
+
+```bash
+curl -X POST -H "X-Vault-Token: $(vault print token)" -H "X-Vault-Request: true" -d '{"policies":["vault-demo-app-policy"],"ttl":"0s","explicit_max_ttl":"0s","period":"0s","display_name":"","num_uses":0,"renewable":true,"type":"service","entity_alias":""}' http://127.0.0.1:8200/v1/auth/token/create
+```
+
+You are now ready to update the config server. In the `vault-config-server` project, edit `src/main/resource/application.yml` to add Vault as the config backend:
+
+```yml
+server:
+  port: 8888
+
+spring:
+  profiles:
+    active: vault
+  cloud:
+    config:
+      server:
+        vault:
+          host: 127.0.0.1
+          port: 8200
+          kvVersion: 2
+logging:
+  level:
+    root: TRACE
+```
+
+Note that the logging level is set to TRACE to see the interaction between the server and Vault. Restart the `vault-config-server`.
+
+```shell
+./mvnw spring-boot:run
+```
+
+You should see the logs below if the server was configured correctly:
+
+```
+2020-05-01 01:19:10.105  INFO 14072 --- [main] SpringBootConfigurationServerApplication:
+ Started SpringBootConfigurationServerApplication in 4.525 seconds (JVM running for 4.859)
+```
+
+You will not see the config server trying to connect to Vault in the logs. That will happen when a config client requests the properties. Start the `vault-demo-app`, passing the token just created in the environment variable `SPRING_CLOUD_CONFIG_TOKEN`. For example:
+
+```shell
+SPRING_CLOUD_CONFIG_TOKEN=hvs.CAESIKd9pYyc9xesiqmwvep... \
+./mvnw spring-boot:run
+```
+
+When the `vault-demo-app` starts, it will request the configuration to the config server, which in turn will make a REST to Vault. In the config server logs, with enough logging level, you will be able to see:
+
+```shell
+2022-09-09 19:21:57.778 DEBUG 12359 --- [nio-8888-exec-1] ...: Received [GET /vault-demo-app/dev HTTP/1.1
+Accept: application/vnd.spring-cloud.config-server.v2+json
+X-Config-Token: hvs.CAESIKd9pYyc9xesiqmwvep...
+User-Agent: Java/17
+Host: localhost:8888
+Connection: keep-alive
+
+]
+```
```

Comment:
I see the following with no logging changed. This seems like it's good enough:

```
2022-09-23 16:48:06.520  INFO 9092 --- [           main] o.s.c.c.c.ConfigServerConfigDataLoader   : Fetching config from server at : http://localhost:8888
2022-09-23 16:48:06.520  INFO 9092 --- [           main] o.s.c.c.c.ConfigServerConfigDataLoader   : Located environment: name=vault-demo-app, profiles=[dev], label=null, version=null, state=null
```

Reply:
That' s from the app, I want to show when the config server talks with vault. I fixed the log level and the logs example.